test/ITensorLegacyMPS/base/test_mps.jl

```diff
@@ -856,6 +856,13 @@ end
     @test res[2, 1] ≈ eSx[3:7]
     @test res[1, 2] ≈ eSx[3:7]
     @test res[2, 2] ≈ eSz[3:7]
+
+    # Test that passing zero-norm MPS leads to an error
+    # (expect not well-defined in that case)
+    psi0 = copy(psi)
+    psi0[1] *= 0.0
```

Comment:
```suggestion
    psi0[1] *= false
```
Generally we are trying to be more agnostic about data types when possible, `false` is the lowest zero type so promotes up to any element type `psi0` would have. It wouldn't change this test but I think it's a good habit to get into.

Reply:
Just made that change. I think it makes sense for library code to be able to be generic, but I'd feel nervous to do this too much in code that users might read, because if I didn't know the reason from the above, I would have not understood that `false` was behaving as a number type here.

Reply:
Sounds good. Also there are probably better patterns for this kind of thing like `zero(scalartype(psi0))`. In general I agree we should think carefully about the readability of generic code.